### PR TITLE
added tribool behaviour to HelpCommand.verify_checks

### LINKED ISSUE #2008: HelpCommand.VerifyChecks Tribool - Always, only in guild context, never
### The Problem

I would like it so I can achieve the following without having to make my own help implementation if possible:

 - If a user triggers help from a guild, the bot will verify checks when generating help (send to DMs or context dependant on option still)
 - If a user triggers help from DMs, the bot will ignore checks and show all commands

### The Ideal Solution

Make [HelpCommand.verify_checks](https://discordpy.readthedocs.io/en/rewrite/ext/commands/api.html#discord.ext.commands.HelpCommand.verify_checks) a Tribool with the following occurring:

 - True: Always verify checks
 - None: Only verify in guild context
 - False: Never verify

### The Current Solution

I'm trying to avoid using guild only checks and instead, writing them as part of the command so that they will still show in help if triggered from DMs.

### Summary

HelpCommand.VerifyChecks Tribool - Always, only in guild context, never

### PR DESCRIPTION
**Pull request by sudosnok**
## Summary
Allows verify_checks to be None, giving finer control over when check are run
Closes #2008 

## Checklist

- [x] If code changes were made then they have been tested.
    - [x] I have updated the documentation to reflect the changes.
- [ ] This PR fixes an issue.
- [x] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

File: discord/ext/commands/help.py [hunk 1]
```diff
@@ -272,7 +272,7 @@ class HelpCommand:
     show_hidden: :class:`bool`
         Specifies if hidden commands should be shown in the output.
         Defaults to ``False``.
-    verify_checks: :class:`bool`
+    verify_checks: Union[:class:`bool`, :class:`None`]
```
Comment (Rapptz) [inline discord/ext/commands/help.py:275]:
```suggestion
    verify_checks: Optional[:class:`bool`]
```

The ``None`` behaviour is completely undocumented.

**Review by Rapptz — CHANGES_REQUESTED** [review]

File: discord/ext/commands/help.py (hunk 1 above)
Comment (sudosnok) [reply discord/ext/commands/help.py:275]: No worries, was uhming and ahing about how to document a tribool

**Comment by NCPlayz** [comment]
I generally feel like this tribool should be an enum as it is easier to document the behaviour in a readable manner, as well as being easy to understand the behaviour internally.

File: discord/ext/commands/help.py [hunk 2]
```diff
@@ -272,7 +272,7 @@ class HelpCommand:
     show_hidden: :class:`bool`
         Specifies if hidden commands should be shown in the output.
         Defaults to ``False``.
-    verify_checks: :class:`bool`
+    verify_checks: Optional[:class:`bool`]
         Specifies if commands should have their :attr:`.Command.checks` called
         and verified. Defaults to ``True``.
```
Comment (Rapptz) [inline discord/ext/commands/help.py:277]: The `None` behaviour is still not documented.

**Review by Rapptz — CHANGES_REQUESTED** [review]

File: discord/ext/commands/help.py [hunk 3]
```diff
@@ -274,7 +274,11 @@ class HelpCommand:
         Defaults to ``False``.
     verify_checks: Optional[:class:`bool`]
         Specifies if commands should have their :attr:`.Command.checks` called
-        and verified. Defaults to ``True``.
+        and verified. If True; always calls :attr:`.Commands.checks`.
+        If None; only calls :attr:`.Commands.checks` in a guild setting.
+        If False; never calls :attr:`.Commands.checks`. Defaults to ``True``.
```
Comment (Rapptz) [inline discord/ext/commands/help.py:279]:
```suggestion
        and verified. If ``True``, always calls :attr:`.Commands.checks`.
        If ``None``, only calls :attr:`.Commands.checks` in a guild setting.
        If ``False``, never calls :attr:`.Commands.checks`. Defaults to ``True``.
```

**Review by Rapptz — CHANGES_REQUESTED** [review]
Sorry, some other minor nits.

**Review by Rapptz — APPROVED** [review]

**Pull request merged**